examples/apps/color_finder.py

```diff
@@ -0,0 +1,702 @@
+        '''
+        above_blob_key = self.keys.at(i, j-1).value
+        self.blobs_dict[above_blob_key].append((i, j))
+        self.keys.at(i, j).set(above_blob_key)
+
+    def join_blob_left(self, i, j):
+        '''Adds current point to the blob to the left.
+
+        Args:
+            i (int): the x-coordinate in self.matrix
+            j (int): the y-coordinate in self.matrix
+        '''
+        left_blob_key = self.keys.at(i-1, j).value
+        self.blobs_dict[left_blob_key].append((i, j))
+        self.keys.at(i, j).set(left_blob_key)
+
+    def merge_up_and_left_blobs(self, i, j):
+        '''Adds current point and points from the above blob into left blob, 
+        then removes the above blob from self.blob_dict
+
+        Args:
+            i (int): the x-coordinate in self.matrix
+            j (int): the y-coordinate in self.matrix
+        '''
+        above_blob_key = self.keys.at(i, j-1).value
+        left_blob_key = self.keys.at(i-1, j).value
+        above_blob_points = self.blobs_dict[above_blob_key]
+        left_blob_points = self.blobs_dict[left_blob_key]
+        for point in above_blob_points:
+            self.blobs_dict[left_blob_key].append(point)
+        self.blobs_dict[left_blob_key].append((i, j))
+        self.keys.at(i, j).set(left_blob_key)
+        for (x, y) in above_blob_points:
+            self.keys.at(x, y).set(left_blob_key)
+        self.blobs_dict.pop(above_blob_key)
+
+    def filter_blobs_dict_by_size(self, n):
+        '''Filters out small blobs from self.blobs_dict.
+
+        Args:
+            n (int): the number of points required of a blob to stay in self.blobs_dict
+        '''
+        self.blobs_dict = dict((blob, list_of_points) for blob, list_of_points in self.blobs_dict.items() if len(list_of_points) >= n)
+
+    def get_largest_blob_with_color(self, color):
+        '''Finds the largest blob of a particular color
+
+        Args:
+            color (string): the desired color of the blob
+
+        Returns:
+            (int, string) specifying the key of the largest blob with that color, or None if no such blob exists
+        '''
+        filtered_dict = dict(((n, k), v) for (n, k), v in self.blobs_dict.items() if k == color)
+        values = filtered_dict.values()
+        if len(values) > 0:
+            longest_blob_list = functools.reduce(lambda largest, current: largest if (largest > current) else current, values)
+            sample_x, sample_y = longest_blob_list[0]
+            return self.keys.at(sample_x, sample_y).value
+        else:
+            return None
+
+    def make_blob_centers(self):
+        '''Constructs a dictionary to keep track of the center of each blob.
+
+        Returns:
+            dict of blob centers
+                Key: color (string): the color of the blob
+                Value: x, y (int, int): the approximate center of the blob as coordinates of self.matrix
+        '''
+        info = {}
```

Comment:
given that info is "dict of blob centers" I suggest renaming the variable to blob_centers